### PR TITLE
Map tags #29

### PR DESCRIPTION
**Pull request by rozetko**
Added tagsSchema, updated examples and docs.
https://github.com/CorpGlory/csv2influx/issues/29

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,10 +27,13 @@
   },
   "homepage": "https://github.com/CorpGlory/csv2influx#readme",
   "dependencies": {
+    "chai": "^4.1.2",
```
Comment (jonyrock) [inline package.json]: it should be in dev-dependencies

File: package.json [hunk 2]
```diff
@@ -27,10 +27,13 @@
...
     "csv-parse": "^1.2.4",
     "influxdb-nodejs": "^2.7.4",
+    "mocha": "^4.0.1",
```
Comment (jonyrock) [inline package.json]: it should be in dev-dependencies

File: package.json [hunk 3]
```diff
@@ -27,10 +27,13 @@
...
     "progress": "^2.0.0",
-    "stream-transform": "^0.1.2"
+    "stream-transform": "^0.1.2",
+    "test": "^0.6.0"
```
Comment (jonyrock) [inline package.json]: not sure we really need it

File: test/importer.js [hunk 4]
```diff
@@ -5,19 +5,23 @@ describe("Importer", function() {
   describe("parseValue", function() {
     it("Parses datetes to nanoseconds", function() {
```
Comment (jonyrock) [inline test/importer.js:6]: `datetes` -> `dates` ok, I know that I made that mistake )

File: importer.js [hunk 5]
```diff
@@ -224,6 +219,15 @@ class Importer {
     }
   }
 
+  _convertSchemaToObject(schema, namesMapping, record) {
+    var obj = {};
+
+    for(var key in schema) {
+        obj[key] = record[namesMapping[key]];
```
Comment (jonyrock) [inline importer.js]: bad indent: lets use two spaces

File: README.md [hunk 6]
```diff
@@ -30,32 +30,35 @@ Options:
...
-      },
+      }
+    },
+    "tagSchema": {
       "location": {
         "from": "location",
-        "type": "string"
-      },
-    },
+        "type": "*" // type of tag can be "*" (any value) or array of possible values
```
Comment (jonyrock) [inline README.md:58]: nice one! 
but let's check indent here

File: config.js [hunk 7]
```diff
@@ -56,28 +56,14 @@ function _checkConfigObject(confObj) {
   if(!confObj.mapping) {
     return 'no mapping';
   }
-  if(!confObj.mapping.fieldSchema) {
-    return 'mapping.fieldSchema';
+  if(!confObj.mapping.time) {
+    return 'mapping.time';
```
Comment (jonyrock) [inline config.js]: let's change to "Time field is missing" or something like that

File: config.js [hunk 8]
```diff
@@ -56,28 +56,14 @@ function _checkConfigObject(confObj) {
...
-  if(!timestamp) {
-    return 'no timestamp field';
+  if(!confObj.mapping.time.format) {
+    return 'no format specified for time'
   }
-
-  if(!format) {
-    return 'no format specified at timestamp field'
+  if(!confObj.mapping.fieldSchema) {
+    return 'mapping.fieldSchema';
```
Comment (jonyrock) [inline config.js]: let's make more friendly error here as well

File: test/importer.js [hunk 9]
```diff
@@ -30,13 +34,22 @@ describe("Importer", function() {
...
     
     it("Parses tagsSchema", function() {
       var vpr = importer.flatSchema(mapping.tagsSchema);
-      expect(vpr.schema).to.not.be.undefined('No schema in result');
+      expect(vpr.schema).to.deep.equal({
+        'name': 'string'
+      });
     });
 
     // TODO: write test
```
Comment (jonyrock) [inline test/importer.js]: I think we can remove this

Comment (jonyrock) [inline test/importer.js]: I think we also should add a test to config.js checking something about new tagsSchema

**Review by jonyrock — COMMENTED** [review]
A bit not cool that we mix new functionality like new types and time field with tagsSchema feature

File: README.md (hunk 6 above)
Comment (rozetko) [reply README.md:58]: actually, don't see problems with indent here :(

Comment (jonyrock) [reply README.md:58]: oops

**Pull request merged**